### PR TITLE
rename expectThrows() to assertThrows()

### PR DESCRIPTION
**Pull request by panchenko**
as suggested again in #1496, Hopefully it helps with having a release sooner.

The important part is making a release, it's not a big deal if there is 1 or 2 methods.

### CODE REVIEW

File: src/main/java/org/junit/Assert.java [hunk 1]
```diff
@@ -989,11 +991,14 @@ public static void assertThrows(Class<? extends Throwable> expectedThrowable, Th
...
-     * @param runnable       a function that is expected to throw an exception when executed
+     * @param runnable a function that is expected to throw an exception when executed
+     * @param message the identifying message for the {@link AssertionError} (<code>null</code>
+     * okay)
      * @return the exception thrown by {@code runnable}
      * @since 4.13
      */
-    public static <T extends Throwable> T expectThrows(Class<T> expectedThrowable, ThrowingRunnable runnable) {
+    public static <T extends Throwable> T assertThrows(Class<T> expectedThrowable,
+            ThrowingRunnable runnable, String message) {
```
Comment (kcooney) [inline src/main/java/org/junit/Assert.java]: In JUnit 4.x the message comes first in the assertion methods, so perhaps we should do that to be consistent.

Besides consistency, the lambda have often be multiple lines, so I personally find it easier to read is the lambda is the last argument

**Comment by kcooney** [comment]
@rschmitt Now that JUnit5 is out and stable, we wanted to (belatedly) prepare to release 4.13. Since JUnit5 combined `assertThrows()` and `expectThrows()` into a single method named `assertThrows()` (and we haven't heard concerns) it seems reasonable to merge the two methods in 4.13.. Looping you in since you did so much work to get this functionally added to 4.x.

**Comment by rschmitt** [comment]
@kcooney I think that merging them is fine; it's better to be consistent with JUnit 5.

File: src/main/java/org/junit/Assert.java [hunk 2]
```diff
@@ -1010,16 +1015,22 @@ public static void assertThrows(Class<? extends Throwable> expectedThrowable, Th
...
-                formatClass(expectedThrowable));
-        throw new AssertionError(message);
+        String notThrownMessage = buildPrefix(message) + String
+                .format("expected %s to be thrown, but nothing was thrown",
+                        formatClass(expectedThrowable));
+        throw new AssertionError(notThrownMessage);
+    }
+
+    private static String buildPrefix(String message) {
+        return message != null && message.length() != 0 ? message + " ==> " : "";
```
Comment (stefanbirkner) [inline src/main/java/org/junit/Assert.java]: Can you please replace `==>` with something unobstrusive like a colon `:`. Thanks.

**Comment by stefanbirkner** [comment]
@panchenko Do you have time to squash the commits?

**Comment by panchenko** [comment]
I can, also can be done while [merging](https://help.github.com/articles/configuring-commit-squashing-for-pull-requests/).

**Comment by stefanbirkner** [comment]
I would like to squash and then rebase. Unfortunately this is not supported by Github's web UI.

**Comment by stefanbirkner** [comment]
Merged. Thanks, @panchenko.

**Comment by stefanbirkner** [comment]
I changed the release notes accordingly.

**Pull request merged**